include/boost/compute/algorithm/detail/scan_on_cpu.hpp

```diff
@@ -36,15 +38,32 @@ inline OutputIterator scan_on_cpu(InputIterator first,
 
     typedef typename
         std::iterator_traits<InputIterator>::value_type input_type;
+    typedef typename
+        std::iterator_traits<OutputIterator>::value_type output_type;
 
     const context &context = queue.get_context();
 
     // create scan kernel
     meta_kernel k("scan_on_cpu");
-    k.add_arg<ulong_>("n");
+
+    // Arguments
+    size_t n_arg = k.add_arg<ulong_>("n");
+    size_t init_arg = k.add_arg<output_type>("init");
+
+    if(!exclusive){
+        k <<
+            k.decl<const ulong>("start_idx") << " = 1;\n" <<
```

Comment:
Use Boost.Compute's `ulong_` instead of `ulong` (which is not always available).